### PR TITLE
feat: delimiting modifiers

### PR DESCRIPTION
**Pull request by benlubas**
Parse `---`, `___` and `===` and add tests.

https://github.com/benlubas/rust-norg/pull/1 stacks on this

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -82,6 +82,45 @@ mod tests {
...
+            // Counter Examples
+            "---not a delimiter",
+            "===nope
+                =end",
+            "_",
+            "hi ---",
+        ]
+        .into_iter()
+        .map(|example| example.to_string() + "\n")
+        .map(|str| dbg!(parse(&str)))
```
Comment (vhyrro) [inline src/lib.rs:117]: nit: leftover `dbg!` call :p

**Comment by vhyrro** [comment]
Code looks great to me. If that's alright tomorrow I'll do some extra tests with this stacked on top of the attached modifier code, just to make sure it's all playing along nicely :)

**Comment by benlubas** [comment]
Ugh, I messed up the commit history pretty hard, and I'm too tired to figure it out right now. I will look again when I wake up

**Comment by vhyrro** [comment]
I rebased my PR on top of latest main, so your rebase should be flawless now. `jj op restore` to the last valid point and try again :p

**Comment by benlubas** [comment]
Much better

**Comment by benlubas** [comment]
rolled into #13

**Pull request closed without merging**